src/bmffimage.cpp

```diff
@@ -521,28 +521,26 @@ void BmffImage::parseTiff(uint32_t root_tag, uint64_t length) {
 }
 
 void BmffImage::parseXmp(uint64_t length, uint64_t start) {
-  if (length > 8) {
-    enforce(start <= io_->size(), ErrorCode::kerCorruptedMetadata);
-    enforce(length <= io_->size() - start, ErrorCode::kerCorruptedMetadata);
-
-    long restore = io_->tell();
-    io_->seek(static_cast<long>(start), BasicIo::beg);
+  enforce(start <= io_->size(), ErrorCode::kerCorruptedMetadata);
+  enforce(length <= io_->size() - start, ErrorCode::kerCorruptedMetadata);
 
-    auto lengthSizeT = static_cast<size_t>(length);
-    DataBuf xmp(lengthSizeT + 1);
-    xmp.write_uint8(lengthSizeT, 0);  // ensure xmp is null terminated!
-    if (io_->read(xmp.data(), lengthSizeT) != lengthSizeT)
-      throw Error(ErrorCode::kerInputDataReadFailed);
-    if (io_->error())
-      throw Error(ErrorCode::kerFailedToReadImageData);
-    try {
-      Exiv2::XmpParser::decode(xmpData(), std::string(xmp.c_str()));
-    } catch (...) {
-      throw Error(ErrorCode::kerFailedToReadImageData);
-    }
+  long restore = io_->tell();
+  io_->seek(static_cast<long>(start), BasicIo::beg);
 
-    io_->seek(restore, BasicIo::beg);
+  auto lengthSizeT = static_cast<size_t>(length);
+  DataBuf xmp(lengthSizeT + 1);
+  xmp.write_uint8(lengthSizeT, 0);  // ensure xmp is null terminated!
+  if (io_->read(xmp.data(), lengthSizeT) != lengthSizeT)
```

Comment:
Then it would be good address all the other instances of io_->read() in this file which I'm not inclined to address right now, so this minimal fix is hopefully ok for now?

Reply:
sure, maybe it makes more sense to address such changes as a whole in a different branch 👍